### PR TITLE
V3 client adoption (Part 1)

### PR DESCRIPTION
**Pull request by nictas**

### CODE REVIEW

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            .cloudFoundryClient(createClient(controllerUrl, oAuthClient));
+        if (target != null) {
+            builder.organization(getOrganizationName(target))
+                .space(getSpaceName(target));
+        }
+        return builder.build();
+    }
+
+    private String getOrganizationName(CloudSpace target) {
+        return target.getOrganization()
```
Comment (enchobelezirev) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java]: Above there is a check for the target whether it is null. Is this check needed here?

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            .getName();
+    }
+
+    private String getSpaceName(CloudSpace target) {
+        return target.getName();
```
Comment (enchobelezirev) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java]: Same question as above?

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java [hunk 3]
```diff
@@ -149,6 +155,12 @@
     private RestTemplate restTemplate;
     private CloudSpace target;
 
+    // Both of these will be utilized in a future commit:
```
Comment (enchobelezirev) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java]: If a cherry-pick is needed, is it possible to cherry-pick this commit and not the other and introduce a NPE?

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java [hunk 4]
```diff
@@ -157,12 +169,14 @@ protected CloudControllerRestClientImpl() {
     }
 
     public CloudControllerRestClientImpl(URL controllerUrl, CloudCredentials credentials, RestTemplate restTemplate,
-        OAuthClient oAuthClient, LoggregatorClient loggregatorClient) {
-        this(controllerUrl, credentials, restTemplate, oAuthClient, loggregatorClient, null);
+        OAuthClient oAuthClient, LoggregatorClient loggregatorClient, CloudFoundryOperations v3OperationsClient,
+        CloudFoundryClient v3Client) {
```
Comment (enchobelezirev) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java]: Wouldn't be better if the variables v3* are named without the v3, if we are with the assumption that everything is v3

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java (hunk 1 above)
Comment (nictas) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java]: Fixed.

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java (hunk 2 above)
Comment (nictas) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/adapters/CloudControllerV3ClientFactory.java]: Fixed.

**Pull request merged**